Update readme and issue template.

Unnoen opened the pull request:
Updated README.md to make it prettier and read nicer.

Added forge logo, description, some badges, adjusted titles for better clarity, mention of patreon in the footer, also added a "common links" index.

I also added a "support table", the goal being a clear-cut list of what is supported and also to stop "X version when?"

The issue template has been adjusted to include common placeholders.

All the .md documents have been moved to `/docs` I'm not sure of the stance of this so let me know what layout is wanted, but github will automatically pull docs from there (like displaying readme in repo root) so functionally it's the same.

Unnoen committed:
Start on updating readme files.

Added logo.
Added short description.
Added release and support badges.
Rename titles for further changes.

Unnoen committed:
Update issue template with basic placeholder

Unnoen committed:
Add Discord badge and support table

Unnoen committed:
Move readme files to docs folder. Add Forge banner to docs/assets. Replace readme Forge logo for one with a background.

Unnoen committed:
Relative path fix for banner.

Unnoen committed:
Update Forge banner logo to transparent SVG with blue anvil.

Unnoen committed:
Further update to readme

Add index of common links.
Further clarify sections.
Mention contribution guidelines.
Mention Patreon at footer.

Unnoen committed:
Modify/revert clarity change.

Extra headings aren't needed for such a small readme at the moment.

Unnoen committed:
Add notes segment to readme

Add a small notes segment for notices.

Unnoen commented:
The latest version badge pulls from Forge's maven-metadata.xml so will *not* need to be updated on every build.

The support table would have three types, Development, Maintained and Legacy. I was going to add a key to the table to show what those mean but I feel it's unnecessary, at least at the moment. This *will* need to be manually adjusted, but since it's major versions, it will be very rare.

bs2609 commented on docs/README.md:
```diff
@@ -0,0 +1,66 @@
+
+# Installing Forge
+
+Go to [the Forge website](http://files.minecraftforge.net)
```
`https` links wherever possible would be nice.

bs2609 commented on docs/README.md:
```diff
@@ -0,0 +1,66 @@
+# Installing Forge
+
+Go to [the Forge website](http://files.minecraftforge.net)
+ and select the minecraft version you wish to get forge for from the list.
```
Use 'Minecraft', 'Forge', 'GitHub', etc. throughout, as these are the 'official' names.

bs2609 commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -1,4 +1,16 @@
+
+**Minecraft Version:** {mc version}
+
+**Forge Version:** {forge version}
```
You might need a "**not** the Minecraft version, **DO NOT** specify 'latest'/'recommended'" here.

Unnoen replied:
I don't think it's necessary to mention *not* the Minecraft version as it's mentioned in it's own placeholder directly above.

bs2609 commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -1,4 +1,16 @@
+
+**Forge Version:** {forge version}
+
+**Full Log:** {link to gist/pastebin with full latest.log}
```
For GitHub issues, Gist is probably preferred, as it allows larger file sizes and I think stays valid for longer (the poster can be assumed to have the required account, for obvious reasons).

bs2609 reviewed (commented):
Couple of style points and suggestions:

Unnoen committed:
HTTPS links and capitalise names.

Unnoen committed:
Update ISSUE_TEMPLATE.md

HTTPS link.
Specify version number.
Capitalise names.
Mention gist as preferred host.

0xilly reviewed (commented):
lgtm maybe add a note that maintained means bug fixes only.

marvin-roesch commented:
GitHub supports [multiple issue templates](https://help.github.com/en/articles/about-issue-and-pull-request-templates). I think it's a good idea to provide at least two templates as part of this PR: one for bug reports and one for feature requests (which are rare, but do happen).

A nice bonus of multiple templates is that we get to display a short description of what constitutes an issue fitting the template. This might help reduce the occurrence of issues that really belong in the forums, if the description is properly worded.

Unnoen commented:
> lgtm maybe add a note that maintained means bug fixes only.

"Maintained" was to mean "active". Once 1.14 dropped 1.12 would become "Legacy". If there's a better suggestion for the statuses let me know, but I was suggesting as follows:

**Development**: Work being done in private/Patches (No public build)
**Maintained**: Fully supported and actively receiving work.
**Legacy**: Limited support. Bug fixes only.

Unnoen commented:
> GitHub supports [multiple issue templates](https://help.github.com/en/articles/about-issue-and-pull-request-templates). I think it's a good idea to provide at least two templates as part of this PR: one for bug reports and one for feature requests (which are rare, but do happen).
> 
> A nice bonus of multiple templates is that we get to display a short description of what constitutes an issue fitting the template. This might help reduce the occurrence of issues that really belong in the forums, if the description is properly worded.

Ah yes, I saw that there was a new issue template method but didn't look into it, it does seem very useful. I'll make some for bugs and features and look through existing issues to see if there's anything else that might deserve it's own template.

Unnoen committed:
Update to newer issue template system. Add bug report and feature request templates.

Unnoen commented:
The templates have been added but I'm not sure what you want the content of them to be.

They will auto assign the Bug and Feature label respectively.

At the moment the feature request one is GitHub's default.

Unnoen committed:
Mention flocker store.

mezz commented:
Thanks for doing this, it looks nice!

Some things to note for your new issue templates though:

- The majority of player bug reports should go to the Forge Forums first. The bug tracker is mainly for developers who have narrowed down the source of the bug so they are confident it comes from Forge itself. Using the forums as the front-line for bug reports helps us cut down on "my game crashed" vague reports here, the participants in the forum thread can help narrow down the bug and figure out what's wrong.

- Feature suggestions are rarely implemented. Generally if someone wants a feature, they should be ready to implement it themselves in a PR and we can work from there. If they are not certain about the details, they can open a Draft PR or discuss it in the Discord before starting on it.

With that in mind, please re-think the issue templates to help give reporters the right expectations.

Unnoen committed:
Edit issue templates to direct users to what they should do first.

Unnoen committed:
Merge branch 'readme-update' of https://github.com/Unnoen/MinecraftForge into readme-update

Unnoen commented:
Updated the bug report template to mention the forum for support first, the feature request template now mentions that opening a PR/discussion is the best course of action.

I decided to keep the feature request template as it can act as a "notice" for users who decide to open an issue for something they want implemented. Also, it does have its place, for example https://github.com/MinecraftForge/MinecraftForge/issues/5732 as a placeholder/update location.

Unnoen commented:
*Please note that there are comments in said templates that will only show when viewing raw or in an issue.*

Unnoen committed:
Optimise Forge SVG.

mezz commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,26 @@
+
+<!-- Thank you for reporting! -->
+
+<!-- Please go to the forum first for support with crashes and issues. -->
```
```suggestion
<!-- For support with crashes and issues, please create a post on the forum instead: -->
```

mezz commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,13 @@
+
+---
+
+<!-- Please note, feature requests are rarely implemented. -->
```
```suggestion
<!-- Please note, feature requests that are submitted as GitHub issues here are rarely implemented. -->
```

mezz commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,13 @@
+---
+
+<!-- Please note, feature requests are rarely implemented. -->
+<!-- If you'd like to see something in Forge, you should be ready to implement it yourself. -->
```
```suggestion
<!-- If you'd like to see something in Forge, you should be ready to implement it as a PR. -->
```

mezz reviewed (changes requested):
Looks good. I'd like to improve a few of the messages to be more direct or slightly more friendly sounding so people who really want to contribute are not turned away.

Unnoen committed:
Friendlyfy bug_report.md

Co-Authored-By: James Mitchell <tehgeek@gmail.com>

Unnoen committed:
Clarify GitHub issue in feature_request.md

Co-Authored-By: James Mitchell <tehgeek@gmail.com>

Unnoen committed:
Clarify "as PR" in feature_request.md

Co-Authored-By: James Mitchell <tehgeek@gmail.com>

Unnoen committed:
1.14.2 is out baby! Reflect that in readme.

1.14.2 is maintained and 1.12.2 is legacy now.

Unnoen commented:
Commited the changes proposed and I changed the PR to the 1.14.x branch.

tterrag1098 commented on docs/README.md:
```diff
@@ -0,0 +1,65 @@
+| Version  | Support |
+| ------------- | ------------- |
+| 1.14.2  | Maintained  |
+| 1.12.2  | Legacy  |
```
1.12.2 will not be in legacy support until 1.14 has a recommended build.

Unnoen replied:
Oh whoops, I got ahead of myself. 1.13.2 is no longer being maintained though is it?

Unnoen replied:
Also, should I keep "Maintained"? Maybe "Active (latest)" and "Active (stable)"?

Unnoen committed:
1.12.2 isn't in legacy yet.

Change wording in support table to reflect why multiple versions may be "maintained".

Unnoen commented:
Removed the "legacy" from 1.12.2 and used "Active (latest)" and "Active (stable)" respectively.
